servant-client/src/Servant/Client.hs

```diff
@@ -1,3 +1,14 @@
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE DeriveDataTypeable         #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
```

Comment:
Is there a reason for moving this around? I generally prefer to not have too much code in non-`Internal` modules since then I can export everything for testing etc. without cluttering the haddocks and API.

Reply:
Not really (I planned to make non-IO client, but it's difficult given we have streaming now. I'll revert this change.

For the record, I partially disagree, I want e.g. `requestToClientRequest`to be a part of *stable public API* (I'd rather export it from `Servant.Client`)